cdap-data-fabric/src/main/java/co/cask/cdap/spi/metadata/MetadataMutation.java

```diff
@@ -48,6 +50,14 @@ public MetadataEntity getEntity() {
 
   /**
    * Create or replace the metadata for an entity.
+   *
+   * For any scope that appears in the new Metadata, all existing metadata
+   * is replaced according to the given directives. If a scope does not
+   * occur in the Metadata, existing metadata in that scope will not be changed.
+   *
+   * Note that the typical use for this is the creation (or recreation) of an
+   * entity, when all the entity's system metadata are defined for the first
+   * time, or redefined.
```

Comment:
This class seems to used in a Collection which can be set
```
Collection<MetadataChange> batch(Collection<? extends MetadataMutation> mutations) throws IOException;
```

Should provide equals and hashCode implementation

Reply:
Good point. Added.